content/en/docs/concepts/overview/object-management-kubectl/kustomization.md

```diff
@@ -0,0 +1,752 @@
+resources:
+- deployment.yaml
+images:
+- name: nginx
+  newName: my.image.registry/nginx
+  newTag: 1.4.0
+EOF
+```
+By `kubectl kustomize ./`, you can see that the image being used is updated.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-nginx
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      run: my-nginx
+  template:
+    metadata:
+      labels:
+        run: my-nginx
+    spec:
+      containers:
+      - image: my.image.registry/nginx:1.4.0
+        name: my-nginx
+        ports:
+        - containerPort: 80
+```
+Sometimes, the application running in Pods may need to know configuration from other objects. For example,
+a Pod from a Deployment object need to read the corresponding Service name from Env or as a command argument.
+Since the Service name may change as namePrefix or nameSuffix is added in the `kustomization.yaml` file. It is
+not recommended to hard code the Service name in the command argument. For this usage, Kustomize can inject
+the Service name dynamically into containers through `vars`.
+
+```shell
+# Create a deployment.yaml file
+cat <<EOF > deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-nginx
+spec:
+  selector:
+    matchLabels:
+      run: my-nginx
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        run: my-nginx
+    spec:
+      containers:
+      - name: my-nginx
+        image: nginx
+        command: ["start", "--host", "\$(MY_SERVICE_NAME)"]
+EOF
+
+# Create a service.yaml file
+cat <<EOF > service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-nginx
+  labels:
+    run: my-nginx
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    run: my-nginx
+EOF
+
+cat <<EOF >./kustomization.yaml
+namePrefix: dev-
+nameSuffix: "-001"
+
+resources:
+- deployment.yaml
+- service.yaml
+
+vars:
+- name: MY_SERVICE_NAME
+  objref:
+    kind: Service
+    name: my-nginx
+    apiVersion: v1
+EOF
+```
+By `kubectl kustomize ./`, you can see that the Service name injected into containers is `dev-my-nginx-001`.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dev-my-nginx-001
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      run: my-nginx
+  template:
+    metadata:
+      labels:
+        run: my-nginx
+    spec:
+      containers:
+      - command:
+        - start
+        - --host
+        - dev-my-nginx-001
+        image: nginx
+        name: my-nginx
+```
+
+## Bases and Overlays
+Kustomize has concepts **bases** and **overlays**. A **base** is a directory with a `kustomization.yaml`, which contains a
+set of resources and associated customization. A base could be either a local directory or a directory from a remote repo,
+as long sa a `kustomization.yaml` is present inside. An **overlay** is a directory with a `kustomization.yaml` that refers to other
+kustomization directories as its `bases`. A **base** has no knowledge of an overlay and can be used in multiple overlay.
```

Comment:
as long as

Reply:
updated

Reply:
nit: line 300,
On top of Resources, one can apply different customizations by applying patches.

Reply:
nit: line 512, what usage: substituting environment variables? Are you sure you need `dynamically`?
For this usage, Kustomize can inject the Service name dynamically into containers through `vars`.

Possible edit: 
Sometimes, the application running in a Pod may need to use configuration values from other objects. 

nit: question: This seems to imply live updates?
For example,
a Pod from a Deployment object may need to read the corresponding Service name from Env or as a command argument.

Reply:
nit: s/overlay/overlays at end of sentence

Reply:
I'll remove `dynamically`, it's not live updates.

Reply:
updated